### DISCUSSION TITLE
ApiKey support for securing the Cerbos API?

### DISCUSSION THREAD

**Discussion by anywhere88** [cerbos/cerbos#1626] [Q&A] [answered]
Hello, I was trying to figure out if Cerbos provides an API authentication mechanism (ApiKey?) to avoid uncontrolled access to the API set.
I could not find anything about this and securing the Cerbos instance at a network level seems the only option. Is this correct?

Thanks

**Comment by charithe** [comment, marked as answer]
You're correct. There's no builtin way to restrict access to the check API because it doesn't expose any sensitive data. If you want to do it, one way to do it is to make Cerbos listen on 127.0.0.1 or a Unix domain socket and configuring a proxy that enforces the access restrictions. We have an example of configuring Ghostunnel on a Kubernetes environment here: https://docs.cerbos.dev/cerbos/latest/deployment/k8s-sidecar.html
